fdbclient/include/fdbclient/BulkLoading.h

```diff
@@ -787,17 +802,18 @@ struct BulkLoadJobManifestFileHeader {
 				throw bulkload_manifest_decode_error();
 			}
 			manifestCount = std::stoull(stringRemovePrefix(parts[1], "[ManifestCount]: "));
+			if (!isValid()) {
+				throw bulkload_manifest_decode_error();
+			}
 		} catch (Error& e) {
 			TraceEvent(SevError, "DecodeBulkLoadJobManifestFileHeaderError")
 			    .setMaxEventLength(-1)
 			    .setMaxFieldLength(-1)
 			    .errorUnsuppressed(e)
 			    .detail("CurrentVersion", bulkLoadManifestFormatVersion)
 			    .detail("RawString", rawString);
-			ASSERT(false);
-			// TODO(BulkLoad): cancel bulkload job for this case. The job is not retriable.
+			throw bulkload_manifest_decode_error();
```

Comment:
Should the error have non_retryable in the name? As in bulkload_manifest_decode_non_retryable_error?

Reply:
Will be included in the next PR.